lib/puma/cluster.rb

```diff
@@ -122,10 +129,10 @@ def cull_workers
 
     # @!attribute [r] next_worker_index
     def next_worker_index
-      all_positions =  0...@options[:workers]
-      occupied_positions = @workers.map { |w| w.index }
-      available_positions = all_positions.to_a - occupied_positions
-      available_positions.first
+      occupied_positions = @workers.map(&:index)
+      idx = 0
+      idx += 1 until !occupied_positions.include?(idx)
+      idx
```

Comment:
The reason I changed this method's implementation is because I ran into an edge case while writing tests for this PR: if `@options[:workers]` is decreased (e.g. because the process received a TTOU signal), the previous implementation could potentially return `nil`.